src/components/SignUp/Form.tsx

```diff
@@ -2,17 +2,20 @@ import { useState } from 'react';
 import { useRouter } from 'next/router';
 import { supabase } from '../../utils/supabaseClient';
 
-export const SignUpForm = () => {
+export function SignUpForm() {
 	const [email, setEmail] = useState('');
+	const [username, setUsername] = useState('');
 	const [password, setPassword] = useState('');
 	const router = useRouter();
+
 	const handleSignUp = async () => {
 		const { user, error } = await supabase.auth.signUp({ email, password });
```

Comment:
looks like the username is not being sent over?

Reply:
Signing up will now send additional metadata about the created user. We will need to create a new `profiles` table, a `trigger`, and a `function` to store the user data. You can create all of them at once using the following query:

```sql
create table public.profiles (
  id uuid references auth.users not null,
  username text unique,

  primary key (id)
);

alter table public.profiles enable row level security;

create policy "Public profiles are viewable by everyone."
  on profiles for select
  using ( true );

create policy "Users can insert their own profile."
  on profiles for insert
  with check ( auth.uid() = id );

create policy "Users can update own profile."
  on profiles for update
  using ( auth.uid() = id );

create function public.handle_new_user()
returns trigger
language plpgsql
security definer set search_path = public
as $$
begin
  insert into public.profiles (id, username)
  values (new.id, new.raw_user_meta_data->>'username');
  return new;
end;
$$;

create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();
```

Reply:
@ykdojo Any updates on this?

Reply:
Thank you. My main concern is, what if a username already exists? What would happen, then?

Reply:
In the backend, the `username` column has `UNIQUE` constraint on it. This will make sure that we are never adding a duplicate username to `profiles` table.

On the frontend, we will make a fetch request to the `profiles` table using `username`, if it already exists we will ask the user to choose a different username.

Reply:
Got it, thank you. Mind adding info about how to set it up in README?

Reply:
I am thinking more of creating a script that can be used to setup the whole supabase backend in one go. This will help the contributors be aware of the database structure and will let people who want to self-host a seamless way to do it.

Let me know when you have executed the above SQL query so that we can merge this and move forward to add username validation in a new PR.

Reply:
I think the README update and this one should be merged at the same time so people know how to set it up

Reply:
About the supabase setup script - that's a migration file. And is very much needed, as in the future we can expect DB models modifications, and all of them should be applied through a migration, so everyone is synced, and keep people away from modifying the DB manually.